src/app/grid/grid-validator-service-extended/grid-validator-service-extended.component.ts

```diff
@@ -0,0 +1,94 @@
+import { Component, Directive, Input, ViewChild } from '@angular/core';
+import { AbstractControl, FormGroup, NG_VALIDATORS, ValidationErrors, ValidatorFn, Validators } from '@angular/forms';
+import { IgxGridComponent } from 'igniteui-angular';
+import { employeesData } from '../../data/employeesData';
+
+export function phoneFormatValidator(phoneReg: RegExp): ValidatorFn {
+    return (control: AbstractControl): ValidationErrors | null => {
+        const match = phoneReg.test(control.value);
+        return match ? null : { phoneFormat: { value: control.value } } ;
+    }
+}
+
+@Directive({
+    selector: '[phoneFormat]',
+    providers: [{ provide: NG_VALIDATORS, useExisting: PhoneFormatDirective, multi: true }]
+})
+export class PhoneFormatDirective extends Validators {
+    @Input('phoneFormat')
+    public phoneFormatString = '';
+
+    public validate(control: AbstractControl): ValidationErrors | null {
+        return this.phoneFormatString ? phoneFormatValidator(new RegExp(this.phoneFormatString, 'i'))(control)
+            : null;
+    }
+}
+
+@Component({
+    selector: 'app-grid-validator-service-extended',
+    styleUrls: ['./grid-validator-service-extended.component.scss'],
+    templateUrl: './grid-validator-service-extended.component.html'
+})
+export class GridValidatorServiceExtendedComponent {
+
+    @ViewChild('grid1', { read: IgxGridComponent })
+    public grid: IgxGridComponent;
+
+    public data: any[];
+    public employeesData: any[];
+    public transactionData = JSON.parse(JSON.stringify(employeesData));
+
+    constructor() {
+        this.data = employeesData;
+    }
+
+    public formCreateHandler(formGroup: FormGroup) {
+        const createdOnRecord = formGroup.get('created_on');
+        const lastActiveRecord = formGroup.get('last_activity');
+        createdOnRecord.addValidators(this.futureDateValidator());
+        lastActiveRecord.addValidators([this.pastDateValidator(), this.futureDateValidator()]);
+    }
+
+    public commit() {
+        const invalidTransactions = this.grid.validation.getInvalid();
+        if (invalidTransactions.length > 0) {
+            if (confirm('You\'re commiting invalid transactions. Are you sure?')) {
+                this.grid.validation.clear();
```

Comment:
Since this is called commit it should probably also commit the transactions on confirm and then clear validation:

```
                this.grid.transactions.commit(this.data);
                this.grid.validation.clear();
```
Otherwise it should do nothing.